diff -Naur root-6.10.08/math/mathmore/CMakeLists.txt root-6.10.08.origin/math/mathmore/CMakeLists.txt
--- root-6.10.08/math/mathmore/CMakeLists.txt	2017-10-20 23:19:25.120146670 +0800
+++ root-6.10.08.origin/math/mathmore/CMakeLists.txt	2017-10-20 23:09:14.881595040 +0800
@@ -21,7 +21,8 @@
 
 ROOT_GENERATE_DICTIONARY(G__MathMore ${headers} MODULE MathMore LINKDEF Math/LinkDef.h OPTIONS "-writeEmptyRootPCM")
 
-ROOT_LINKER_LIBRARY(MathMore *.cxx G__MathMore.cxx LIBRARIES ${GSL_LIBRARIES} DEPENDENCIES MathCore BUILTINS GSL)
+ROOT_LINKER_LIBRARY(MathMore *.cxx G__MathMore.cxx LIBRARIES ${GSL_LIBRARIES}
+	DEPENDENCIES MathCore BUILTINS gsl gslcblas)
 
 ROOT_INSTALL_HEADERS()

