diff -Naur root-6.10.08.origin/etc/vmc/Makefile.linux root-6.10.08/etc/vmc/Makefile.linux
--- root-6.10.08.origin/etc/vmc/Makefile.linux	2017-10-21 20:09:59.778194345 +0800
+++ root-6.10.08/etc/vmc/Makefile.linux	2017-10-21 20:13:12.118844304 +0800
@@ -69,7 +69,6 @@
 else
 ifneq (,$(findstring gfortran,$(F77)))
 SHLIB := $(shell $(F77) -print-file-name=libgfortran.so)
-SHLIB += $(shell $(F77) -print-file-name=libgfortranbegin.a)
 else
 SHLIB         = -lg2c
 endif
diff -Naur root-6.10.08.origin/etc/vmc/Makefile.linuxx8664gcc root-6.10.08/etc/vmc/Makefile.linuxx8664gcc
--- root-6.10.08.origin/etc/vmc/Makefile.linuxx8664gcc	2017-10-21 20:09:59.778194345 +0800
+++ root-6.10.08/etc/vmc/Makefile.linuxx8664gcc	2017-10-21 20:13:12.118844304 +0800
@@ -71,7 +71,6 @@
 else
 ifneq (,$(findstring gfortran,$(F77)))
 SHLIB := $(shell $(F77) -print-file-name=libgfortran.so)
-SHLIB += $(shell $(F77) -print-file-name=libgfortranbegin.a)
 else
 SHLIB         = -lg2c
 endif
diff -Naur root-6.10.08.origin/etc/vmc/Makefile.macosx root-6.10.08/etc/vmc/Makefile.macosx
--- root-6.10.08.origin/etc/vmc/Makefile.macosx	2017-10-21 20:09:59.778194345 +0800
+++ root-6.10.08/etc/vmc/Makefile.macosx	2017-10-21 20:13:12.118844304 +0800
@@ -66,6 +66,5 @@
 SHLIB += -L$(shell $(F77) --print-search-dirs | sed -n -e 's/install: //p') -lf95
 else
 SHLIB := $(shell $(F77) -m32 -print-file-name=libgfortran.dylib)
-SHLIB += $(shell $(F77) -m32 -print-file-name=libgfortranbegin.a)
 SHLIB += -ldl
 endif
diff -Naur root-6.10.08.origin/etc/vmc/Makefile.macosx64 root-6.10.08/etc/vmc/Makefile.macosx64
--- root-6.10.08.origin/etc/vmc/Makefile.macosx64	2017-10-21 20:09:59.778194345 +0800
+++ root-6.10.08/etc/vmc/Makefile.macosx64	2017-10-21 20:13:12.118844304 +0800
@@ -66,6 +66,5 @@
 SHLIB += -L$(shell $(F77) --print-search-dirs | sed -n -e 's/install: //p') -lf95
 else
 SHLIB := $(shell $(F77) -m64 -print-file-name=libgfortran.dylib)
-SHLIB += $(shell $(F77) -m64 -print-file-name=libgfortranbegin.a)
 SHLIB += -ldl
 endif
