diff --git a/gen_initramfs.sh b/gen_initramfs.sh
index 955b273..abcdd19 100755
--- a/gen_initramfs.sh
+++ b/gen_initramfs.sh
@@ -777,7 +777,7 @@ print_list()
 append_modules() {
     local group
     local group_modules
-    local MOD_EXT=".ko"
+    local MOD_EXT=".ko*"
 
     print_info 2 "initramfs: >> Searching for modules..."
     if [ "${INSTALL_MOD_PATH}" != '' ]
@@ -821,7 +821,7 @@ append_modules() {
 }
 
 append_drm() {
-    local MOD_EXT=".ko"
+    local MOD_EXT=".ko*"
 
     print_info 2 "initramfs: >> Appending drm drivers..."
     if [ "${INSTALL_MOD_PATH}" != '' ]
diff --git a/gen_moddeps.sh b/gen_moddeps.sh
index 27dab48..1700b35 100755
--- a/gen_moddeps.sh
+++ b/gen_moddeps.sh
@@ -3,7 +3,7 @@
 
 modules_kext()
 {
-    KEXT=".ko"
+    KEXT=".ko.*"
     echo ${KEXT}
 }
 
