--- old/CMakeLists.txt	2018-02-10 12:37:24.069302356 +0800
+++ cbmroot-2017.07/CMakeLists.txt	2018-02-10 15:01:13.141183030 +0800
@@ -217,8 +217,6 @@
 SET(LD_LIBRARY_PATH  ${CBMLIBDIR} ${Boost_LIBRARY_DIRS} ${LD_LIBRARY_PATH})
 
 
-Generate_Version_Info()
-
 # Recurse into the given subdirectories.  This does not actually
 # cause another cmake executable to run.  The same process will walk through
 # the project's entire directory structure.
