diff -Naur --strip-trailing-cr geant-vmc-4.9999.old/cmake/FindROOT.cmake geant-vmc-4.9999/cmake/FindROOT.cmake
--- geant-vmc-4.9999.old/cmake/FindROOT.cmake	2019-03-03 14:33:05.608046709 +0800
+++ geant-vmc-4.9999/cmake/FindROOT.cmake	2019-03-03 14:34:06.959045861 +0800
@@ -97,10 +97,6 @@
 endif()    
 
 if(ROOT_FOUND)
-  # ROOT 6 requires C++11 support
-  if (ROOT_FOUND_VERSION GREATER 59999)
-     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-  endif()
   set(LD_LIBRARY_PATH ${LD_LIBRARY_PATH} ${ROOT_LIBRARY_DIR})
   if(NOT ROOT_FIND_QUIETLY)
     message(STATUS "Found ROOT ${ROOT_VERSION} in ${ROOT_PREFIX}")
