diff -Naur FairRoot-18.0.6.origin/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.cxx FairRoot-18.0.6/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.cxx
--- FairRoot-18.0.6.origin/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.cxx	2019-03-16 15:06:50.832742280 +0800
+++ FairRoot-18.0.6/examples/simulation/Tutorial4/src/mc/FairTutorialDet4.cxx	2019-03-16 15:11:13.645738646 +0800
@@ -361,7 +361,7 @@
     TGeoRotation* rrot = new TGeoRotation("rot",dphi,dtheta,dpsi);
     TGeoCombiTrans localdelta = *(new TGeoCombiTrans(dx,dy,dz, rrot));
 //      localdelta.Print();
-    TGeoHMatrix nlocal = *l3 * localdelta;
+    TGeoHMatrix nlocal = *dynamic_cast<TGeoHMatrix*>(l3) * localdelta;
     TGeoHMatrix* nl3 = new TGeoHMatrix(nlocal); // new matrix, representing real position (from new local mis RS to the global one)
 
     TGeoPhysicalNode* pn3 = gGeoManager->MakePhysicalNode(volPath);
@@ -414,7 +414,7 @@
     TGeoRotation* rrot = new TGeoRotation("rot",dphi,dtheta,dpsi);
     TGeoCombiTrans localdelta = *(new TGeoCombiTrans(dx,dy,dz, rrot));
     localdelta.Print();
-    TGeoHMatrix nlocal = *l3 * localdelta;
+    TGeoHMatrix nlocal = *dynamic_cast<TGeoHMatrix*>(l3) * localdelta;
     TGeoHMatrix* nl3 = new TGeoHMatrix(nlocal); // new matrix, representing real position (from new local mis RS to the global one)
 
     node->Align(nl3);
