diff -Naur FairRoot-16.06b.origin/cmake/modules/FindFairRoot.cmake FairRoot-16.06b/cmake/modules/FindFairRoot.cmake
--- FairRoot-16.06b.origin/cmake/modules/FindFairRoot.cmake	2017-10-22 15:00:11.688872698 +0800
+++ FairRoot-16.06b/cmake/modules/FindFairRoot.cmake	2017-10-22 15:00:56.244143951 +0800
@@ -33,7 +33,7 @@
 )
 
 FIND_PATH(FAIRROOT_LIBRARY_DIR NAMES libBase.so libBase.dylib PATHS
-   ${FAIRROOTPATH}/lib
+  ${FAIRROOTPATH}/lib/FairRoot
   NO_DEFAULT_PATH
 )
 
